server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteStoreStatsIT.java

```diff
@@ -111,29 +173,307 @@ public void testStatsResponseFromLocalNode() {
+                        downloadBytesStarted.add(stats.downloadBytesStarted);
+                        downloadBytesSucceeded.add(stats.downloadBytesSucceeded);
+                        downloadBytesFailed.add(stats.downloadBytesFailed);
+                        downloadBytesAverage.add(stats.downloadBytesMovingAverage);
+                        lastDownloadedSegmentSize.add(stats.lastSuccessfulSegmentDownloadBytes);
+                    }
+                }
+
+                assertEquals(0, uploadsFailed);
+                assertEquals(0, uploadBytesFailed);
+                for (int j = 0; j < response.getSuccessfulShards() - 1; j++) {
+                    assertEquals(uploadsStarted, (long) downloadsStarted.get(j));
+                    assertEquals(uploadsSucceeded, (long) downloadsSucceeded.get(j));
+                    assertEquals(0, (long) downloadsFailed.get(j));
+                    assertEquals(uploadBytesStarted, (long) downloadBytesStarted.get(j));
+                    assertEquals(uploadBytesSucceeded, (long) downloadBytesSucceeded.get(j));
+                    assertEquals(0, (long) downloadBytesFailed.get(j));
+                    assertEquals(uploadBytesAverage, downloadBytesAverage.get(j), 0);
+                    assertEquals(lastUploadedSegmentSize, (long) lastDownloadedSegmentSize.get(j));
+                }
+            }, 45, TimeUnit.SECONDS);
+        }
+    }
+
+    public void testStatsOnShardRelocation() {
+        // Scenario:
+        // - Create index with single primary and single replica shard
+        // - Index documents
+        // - Reroute replica shard to one of the remaining nodes
+        // - Assert that remote store stats reflects the new node ID
+
+        // Create index
+        createIndex(INDEX_NAME, remoteStoreIndexSettings(1, 1));
+        // Index docs
+        indexDocs(true);
+
+        // Fetch current set of nodes in the cluster
+        List<String> currentNodesInCluster = getClusterState().nodes()
+            .getDataNodes()
+            .values()
+            .stream()
+            .map(DiscoveryNode::getId)
+            .collect(Collectors.toList());
+        DiscoveryNode[] discoveryNodesForIndex = client().admin().cluster().prepareSearchShards(INDEX_NAME).get().getNodes();
+
+        // Fetch nodes with shard copies of the created index
+        List<String> nodeIdsWithShardCopies = new ArrayList<>();
+        Arrays.stream(discoveryNodesForIndex).forEach(eachNode -> nodeIdsWithShardCopies.add(eachNode.getId()));
+
+        // Fetch nodes which does not have any copies of the index
+        List<String> nodesIdsWithoutShardCopy = currentNodesInCluster.stream()
+            .filter(eachNode -> !nodeIdsWithShardCopies.contains(eachNode))
+            .collect(Collectors.toList());
+        assertEquals(1, nodesIdsWithoutShardCopy.size());
+
+        // Manually reroute shard to a ndoe which does not have any shard copy at present
```

Comment:
nit - node

Reply:
Resolved. Changed to `nodeIdsWithoutShardCopy`